src/Nix/String.hs

```diff
@@ -231,6 +233,16 @@ runWithStringContextT :: Monad m => WithStringContextT m Text -> m NixString
 runWithStringContextT (WithStringContextT m) =
   uncurry NixString <$> runWriterT m
 
+-- | Run an action that manipulates nix strings, and collect the contexts encountered.
+-- Warning: this may be unsafe, depending on how you handle the resulting context list.
```

Comment:
Is it possible to elaborate a little on what unsafe means, and what handling would cause bad behavior?

Reply:
Well, as soon as you separate the context from the string, you are responsible of doing something "principled" with both the context and the string.

runWithStringContext ensures that the monadic context is merged in a nix string in the end. So you are sure that the contexts you abstracted away end up somewhere sensible. Look at "hnix/src/Nix/{Json,XML}.hs". They take a nix value, extract the strings with `extractNixString` and the WithStringContext ensures that the resulting string gets wrapped in all the contexts that have been encountered during the computation.

This version however makes it possible to just discard these contexts from the resulting tuple.

Any idea how to better phrase the comment ?

Reply:
That sounds good to me, I guess these are going to be be quite infrequently used by an end user anyway!

Reply:
I hope so :-).

In fact this one should probably move to the #554 PR, which will probably be the sole usage for it.
{Json,XML}.hs are the sole usages for WithStringContext. Not really a common tool.

Reply:
Sure, no bother either way :)

Reply:
Moved to derivationStrict PR :-)